docs/GettingStartedDocs/IntelSDKPortingGuide.md

```diff
@@ -0,0 +1,214 @@
+```
+
+Now generate the ECall/OCall bridge/proxy routines for the enclave.
+
+```cmake
+add_custom_command(
+  OUTPUT MyEnclave_t.h MyEnclave_t.c MyEnclave_args.h
+  DEPENDS ${CMAKE_SOURCE_DIR}/MyEnclave.edl
+  COMMAND openenclave::oeedger8r --trusted ${CMAKE_SOURCE_DIR}/MyEnclave.edl
+          --search-path ${OE_INCLUDEDIR})
+```
+
+Finally, build the enclave as an executable. Please don't forget to include `MyEnclave_t.c` (generated above) in the source list.
+
+```cmake
+add_executable(MyEnclave MyEnclave.cpp ${CMAKE_CURRENT_BINARY_DIR}/MyEnclave_t.c)
+
+# Current API version
+target_compile_definitions(MyEnclave PUBLIC OE_API_VERSION=2)
+
+# Needed by the generated MyEnclave_t.c
+target_include_directories(MyEnclave PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+
+target_link_libraries(MyEnclave openenclave::oeenclave openenclave::oelibc)
+```
+
+Enclaves must be signed before they can be loaded. How to sign an enclave depends on how the signing private key is managed in your project. For demonstration purpose, the following snippet generates a random RSA key pair to sign the enclave. An enclave configuration file (`MyEnclave.conf` below) may also be provided. Details of enclave configuration/settings are described later in this document.
+
+```cmake
+# Generate key
+add_custom_command(
+  OUTPUT private.pem public.pem
+  COMMAND openssl genrsa -out private.pem -3 3072
+  COMMAND openssl rsa -in private.pem -pubout -out public.pem)
+
+# Sign enclave
+add_custom_command(
+  OUTPUT MyEnclave.signed
+  DEPENDS MyEnclave MyEnclave.conf private.pem
+  COMMAND openenclave::oesign sign -e $<TARGET_FILE:MyEnclave>
+          -c ${CMAKE_SOURCE_DIR}/MyEnclave.conf -k private.pem)
+```
+
+Audiences are encouraged to look for complete examples under */path/to/OpenEnclave_InstallDir*/share/openenclave/samples directory.
+
+## Migrate Enclave Settings
+
+**Enclave Settings**, also known as **Enclave Metadata**, refer to information consumed by enclave loaders to instantiate enclaves, such as heap size, stack size, number of trusted hardware threads (i.e. number of TCS's), etc.
+
+Enclave settings are specified in human readable text formats by enclave developers, referred to as *configuration files*. Both Intel SGX SDK and OpenEnclave SDK provide tools to compile configuration files into their binary form and embed them into the final enclave image. However, they differ in both format and feature set.
+
+### Configuration File Formats
+
+Intel SGX SDK adopted an XML format for encoding enclave settings in text form, which is usually named as *Enclave*.config.xml. The signing tool (i.e. `sgx_sign`) converts it into binary form and stores it in a dedicated section (i.e. `.sgxmeta`) of the enclave's ELF image before it calculates the enclave's measurement (i.e. `SIGSTRUCT::MRENCLAVE`). At runtime, the ELF section `.sgxmeta` is consumed by the enclave loader to instatiate the exact enclave that matches the measurement caculated by the signing tool.
```

Comment:
```suggestion
The Intel SGX SDK adopted an XML format for encoding enclave settings in text form, which is usually named as *Enclave*.config.xml. The signing tool (i.e., `sgx_sign`) converts it into binary form and stores it in a dedicated section (i.e. `.sgxmeta`) of the enclave's ELF image before it calculates the enclave's measurement (i.e., `SIGSTRUCT::MRENCLAVE`). At runtime, the ELF section `.sgxmeta` is consumed by the enclave loader to instantiate the exact enclave that matches the measurement calculated by the signing tool.
```
Also please split into multiple source lines, as you can see that github has a hard time correctly showing the diffs with such a long line, which makes it harder to review.